atlasdb-config/src/test/java/com/palantir/atlasdb/memory/InMemoryAsyncAtlasDbFactoryTest.java

```diff
@@ -50,29 +50,21 @@ public void asyncInitKvsAsynchronous() {
     }
 
     @Test
-    public void syncInitTimestampServiceSynchronous() {
-        TimestampService timestampService = factory.createManagedTimestampService(null, Optional.empty(), false);
-        assertThat(timestampService.isInitialized()).isTrue();
-        assertThat(timestampService.getFreshTimestamp()).isEqualTo(1L);
-    }
-
-    @Test
-    public void asyncInitTimestampServiceWithReadyKvsSynchronous() {
+    public void syncInitTimestampService() {
         KeyValueService kvs = createRawKeyValueService(false);
-        TimestampService timestampService = factory.createManagedTimestampService(kvs, Optional.empty(), true);
+        TimestampService timestampService = factory.createManagedTimestampService(kvs, Optional.empty(), false);
         assertThat(timestampService.isInitialized()).isTrue();
         assertThat(timestampService.getFreshTimestamp()).isEqualTo(1L);
     }
 
     @Test
-    public void asyncInitTimestampServiceWithAsyncKvsAsynchronous() {
-        KeyValueService kvs = createRawKeyValueService(true);
+    public void asyncInitTimestampService() {
+        KeyValueService kvs = createRawKeyValueService(false);
         TimestampService timestampService = factory.createManagedTimestampService(kvs, Optional.empty(), true);
-
         assertThat(timestampService.isInitialized()).isFalse();
         assertThatThrownBy(timestampService::getFreshTimestamp).isInstanceOf(NotInitializedException.class);
```

Comment:
oof, looks like this assertion failed on CircleCI :/ would it make sense to pass the executor for async init in?

Reply:
unfortunately the race condition here looks hard to avoid

Reply:
(I realise this kind of went out of the scope of the original change, if you'd prefer someone on the team to look just let me know)

Reply:
Yeah I noticed this when making this change. I was surprised the KVS test hasn't flaked before.

I'm happy to fix this. Like you said, the race condition here is hard to avoid. Do you have a suggestion? These tests don't seem that useful to me, I'd probably just remove them.

Reply:
Or we can remove just the not initialized assertions.